woocommerce-services.php

```diff
@@ -668,56 +668,74 @@ public function localize_and_enqueue_service_script( $method_id, $instance_id =
+			foreach ( $labels as $label ) {
+				$carrier = esc_html( $label['carrier_id'] );
+				$tracking = esc_html( $label['tracking'] );
+				$refunded = array_key_exists( 'refund', $label );
 
-            echo '<table class="td" cellspacing="0" cellpadding="6" style="margin-top: 10px; width: 100%; font-family: \'Helvetica Neue\', Helvetica, Roboto, Arial, sans-serif;">';
-            echo '<thead>';
-            echo '<tr>';
-            echo '<th class="td" scope="col">' . __( 'Provider', 'woocommerce-services' ) . '</th>';
-            echo '<th class="td" scope="col">' . __( 'Tracking number', 'woocommerce-services' ) . '</th>';
-            echo '</tr>';
-            echo '</thead>';
-            echo '<tbody>';
-            echo '<tr>';
+				if ( $refunded ) {
+					continue;
+				}
 
-            foreach ( $labels as &$label ) {
-                $label = $label[0];
-                $carrier = $label['carrier_id'];
-                $tracking = $label['tracking'];
+				$markup .= '<tr>';
+				$markup .= '<td class="td" scope="col">' . strtoupper( $carrier ) . '</td>';
 
-                echo '<td class="td" scope="col">' . strtoupper($carrier) . '</td>';
+				switch ( $carrier ) {
+					case "fedex":
+						$tracking_url = 'https://www.fedex.com/apps/fedextrack/?action=track&tracknumbers=' . $tracking;
+						break;
+					case "usps":
+						$tracking_url = 'https://tools.usps.com/go/TrackConfirmAction.action?tLabels=' . $tracking;
+						break;
+				}
 
-                switch ( $carrier ) {
-                    case "fedex":
-                        $tracking_url = 'https://www.fedex.com/apps/fedextrack/?action=track&tracknumbers=' . $tracking;
-                        break;
-                    case "usps":
-                        $tracking_url = 'https://tools.usps.com/go/TrackConfirmAction.action?tLabels=' . $tracking;
-                        break;
-                }
+				$markup .= '<td class="td" scope="col">';
+				$markup .= '<a href="' . esc_html( $tracking_url ) . '">' . $tracking . '</a>';
+				$markup .= '</td>';
+				$markup .= '</tr>';
+			}
 
-                echo '<td class="td" scope="col"><a href="' . $tracking_url . '">' . $tracking . '</a></td>';
-            }
+			// Abort if all labels are refunded
+			if ( empty( $markup ) ) {
+				return;
+			}
 
-            echo '</tr>';
-            echo '</tbody>';
-            echo '</table>';
-        }
+			echo '<h2>' . __( 'Tracking', 'woocommerce-services' ) . '</h2>';
+			echo '<table class="td" cellspacing="0" cellpadding="6" style="margin-top: 10px; width: 100%; font-family: \'Helvetica Neue\', Helvetica, Roboto, Arial, sans-serif;">';
+			echo '<thead>';
+			echo '<tr>';
+			echo '<th class="td" scope="col">' . __( 'Provider', 'woocommerce-services' ) . '</th>';
+			echo '<th class="td" scope="col">' . __( 'Tracking number', 'woocommerce-services' ) . '</th>';
+			echo '</tr>';
+			echo '</thead>';
+			echo '<tbody>' . $markup . '</tbody>';
+			echo '</table>';
```

Comment:
This is optional, but recommended - we mix PHP and HTML to avoid a lot of unnecessary `echo`s. Here's an example: https://github.com/Automattic/woocommerce-services/blob/master/woocommerce-services.php#L1023-L1029